### PR TITLE
HIVE-24523: Vectorized read path for LazySimpleSerde does not honor the SERDEPROPERTIES for timestamp

### PR DESCRIPTION
**Pull request by nareshpr**
**What changes were proposed in this pull request?**
To include user provided serde timestampFormats in tableScan

**Why are the changes needed?**
Without the changes, user data with custom timestampFormats are returned as NULL

**Does this PR introduce any user-facing change?**
No

**How was this patch tested?**
Added a new testcase which reproduce the issue without my fix.

### CODE REVIEW

File: serde/src/java/org/apache/hadoop/hive/serde2/lazy/fast/LazySimpleDeserializeRead.java [hunk 1]
```diff
@@ -350,7 +350,12 @@ public LazySimpleDeserializeRead(TypeInfo[] typeInfos,
       throw new RuntimeException("serialization.last.column.takes.rest not supported");
     }
 
-    timestampParser = new TimestampParser();
+    List<String> timestampFormats = lazyParams.getTimestampFormats();
+    if (timestampFormats != null) {
```
Comment (deniskuzZ) [inline serde/src/java/org/apache/hadoop/hive/serde2/lazy/fast/LazySimpleDeserializeRead.java:354]: minor: i would use ternary operator here

**Review by deniskuzZ — APPROVED** [review]
LGTM, just 1 minor comment to consider

File: serde/src/java/org/apache/hadoop/hive/serde2/lazy/fast/LazySimpleDeserializeRead.java (hunk 1 above)
Comment (nareshpr) [reply serde/src/java/org/apache/hadoop/hive/serde2/lazy/fast/LazySimpleDeserializeRead.java]: Thanks for the review @deniskuzZ.
I updated the code with ternary operator.

**Comment by deniskuzZ** [comment]
hi @nareshpr, could you please fill in description in the pull request, so I could merge it.

**Comment by nareshpr** [comment]
Done

**Pull request merged**